modules/plot/main/coffee/Axis.coffee

```diff
@@ -321,60 +318,20 @@ class Axis
       @yScale.domain(domain)
     else
 
-      # OPTIM: this expensive calculation is not needed if the scales are non auto
-      # XXX: band series?
-
-      ymin = undefined
-      ymax = undefined
-
-      types = hx.groupBy(@series(), (d) -> d._.type)
-      stackGroups = types.map (d) ->
-        type: d[0]
-        group: hx.groupBy(d[1], (s) -> if supportsGroup(s) then s.group() else undefined)
-
-      for type in stackGroups
-        for group in type.group
-          series = group[1]
-          if group[0] == undefined
-
-            ys = for s in @series()
-              data = s.data()
-              if s instanceof StraightLineSeries
-                if not data.dx and not data.dy and data.y
-                  [data.y, data.y]
-                else
-                  undefined
-              else if s instanceof BandSeries
-                extent2(data, ((d) -> d.y1), (d) -> d.y2)
-              else
-                extent(data, (d) -> d.y)
-            ys = ys.filter((d) -> d?)
+      { yMinMightBeAuto, yMaxMightBeAuto } = { yMaxMightBeAuto: @y.max(), yMinMightBeAuto: @y.min() }
```

Comment:
Not sure I'm keen on destructuring like this - you may as well do `yMinMightbeAuto = @y.min()`